docs/develop/webapps.md

```
@@ -266,6 +266,26 @@ As the cookie is set to be [`HttpOnly`](https://developer.mozilla.org/en-US/docs
+const here =
+  window.location.pathname + window.location.search + window.location.hash
+window.location.href = '/admin/#/login?redirect=' + encodeURIComponent(here)
+```
+
+After a successful form login or OIDC sign-in the browser is sent back to that URL. The session cookie is in place by then, so the webapp's subsequent requests are authenticated.
+
+The `redirect` value must be a same-origin relative URL: it must start with `/` (but not `//`), contain no backslashes or control characters, and not point at the admin login route itself. Invalid values are silently ignored and the user lands on the dashboard after login.
+
+For OIDC-only deployments, a webapp can skip the admin UI and link directly to the OIDC endpoint, which accepts the same parameter:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)